HIP/hip-000.md

```diff
@@ -0,0 +1,175 @@
+
+Any entity wishing to operate a gRPC-Web proxy node would request to have that node added to the network address book via a `nodeCreate` (which, initially, requires council approval), and would manage that node via `nodeUpdate` as needed to maintain current and up-to-date information for the service endpoints of that node.
+
+### **Protobufs**
+
+Add a new field to `NodeCreateTransactionBody`.
+
+```protobuf
+message NodeCreateTransactionBody {
+...
+    /**
+     * An administrative key controlled by the node operator.
+     */
+    proto.Key admin_key = 7;
+
+    **/**
+     * A web proxy for gRPC from non-gRPC clients.
+     */
+    proto.ServiceEndpoint grpc_proxy_endpoint = 8;**
+}
+```
+
+Add a new field to `NodeUpdateTransactionBody`.
+
+```protobuf
+message NodeUpdateTransactionBody {
+...
+    /**
+     * An administrative key controlled by the node operator.
+     */
+    proto.Key admin_key = 8;
+
+    **/**
+     * A web proxy for gRPC from non-gRPC clients.
+     */
+    proto.ServiceEndpoint grpc_proxy_endpoint = 9;**
+}
+
+```
+
+Partial detail for updates to the `Node` message in state.
+
+```protobuf
+/**
+ * A single address book node in the network state.
+ */
+message Node {
+...
+    /**
+     * An administrative key controlled by the node operator.
+     */
+    proto.Key admin_key = 10;
+
+    **/**
+     * A web proxy for gRPC from non-gRPC clients.
+     */
+    proto.ServiceEndpoint grpc_proxy_endpoint = 11;**
+}
+
+```
+new NodeCreateTransaction()
+	....
+	.setNodeType ()
+	...
+
+## Backwards Compatibility
+
+Initially, there should be no change or impact to existing users of the SDKs and gRPC-Web nodes.
+The current set of nodes will remain available at the current addresses. After all SDKs are migrated to use dynamic lookup of the addresses via the mirror node APIs, the existing nodes are expected to remain at the current addresses for some time, but may, eventually, begin to migrate.  Clients using SDKs should adopt the new SDK versions that support dynamic address lookup within this time period.  Clients not using SDKs to access the gRPC-Web nodes should also migrate to look up the node addresses via the mirror node APIs in the same time period.
+Any client that chooses to continue using the legacy approach of manually maintaining a fixed static list of gRPC-Web nodes may continue to do so, with the constraint that those entities may need to make more frequent updates to their static lists, but the information for those updates will be available from the mirror node.
+
+## Security Implications
+
+If there are security concerns in relation to the HIP, those concerns should be explicitly addressed to make sure reviewers of the HIP are aware of them.
+
+## How to Teach This
+
+Address Book documentation will need to be updated to reflect the new field.  
```

Comment:
This seems more like a backwards compatibility item, rather than recommended approach to explaining this HIP to someone new to the Hedera ecosystem.